src/libsystemd/sd-journal/journal-file.c

```diff
@@ -4082,6 +4082,53 @@ static void journal_default_metrics(JournalMetrics *m, int fd, bool compact) {
                   m->n_max_files);
 }
 
+static uint64_t get_compress_threshold_bytes(uint64_t compress_threshold_bytes) {
+        return compress_threshold_bytes == UINT64_MAX ?
+                DEFAULT_COMPRESS_THRESHOLD :
+                MAX(MIN_COMPRESS_THRESHOLD, compress_threshold_bytes);
+}
+
+static int set_metrics(JournalFile *f, JournalMetrics *metrics, JournalFile *template) {
+        assert(f);
+        int r;
+
+        if (journal_file_writable(f)) {
```

Comment:
maybe invert the check and return early here. we generally prefer early exit over deep nesting